js/runtime.js

```diff
@@ -572,16 +566,40 @@
         },
 
         shape: {
-            fillShape: function(shp){
-                shp();
+            fill: function(pathSet){
```

Comment:
Yeah, I think the `Shape` abstraction is great, and the fact that it's just a `drawable` makes it really easy to just do the proper context stuff and fill/stroke. I think `Shape`'s `draw` method should probably be doing most of the work here.